### PR TITLE
Port I2C fixes from STM32F1xx HAL.

### LINKED ISSUE #31: BlockingI2c does not send ACK after read
When trying to read bytes from a slave with .read(), the master never sends an ACK by pulling SDA low after every succesfully byte read. This causes the slave to hang and eventually timeout. The same behaviour is observed with 1 byte reads as well.
As can be seen from the included image, the bahaviour is incorrect and SDA is not pulled low on the 9th clock cycle. Only the data portion of the signal is shown. The data is the same in both images; 0x73.

Image Pasted at 2020-11-24 23-03

The issue can be reproduced on a longan nano (GD32VF103) with the following code, with of course a suitable slave device connected and addresses substituted. 

```
    let periph = pac::Peripherals::take().unwrap();
    let mut rcu = periph
        .RCU
        .configure()
        .ext_hf_clock(8.mhz())
        .sysclk(108.mhz())
        .freeze();

    let mut afio = periph.AFIO.constrain(&mut rcu);
    let gpiob = periph.GPIOB.split(&mut rcu);
    let i2c_0 = periph.I2C0;
    let mut delay = McycleDelay::new(&rcu.clocks);
    delay.delay_ms(2);

    let scl = gpiob.pb8.into_alternate_open_drain();
    let sda = gpiob.pb9.into_alternate_open_drain();
    
    let mut i2c_handle = BlockingI2c::i2c0 (
        i2c_0,
        (scl, sda),
        &mut afio,
        Mode::standard(100.khz()),
        &mut rcu,
        998,
        1,
        998,
        998
    );

    let mut read_buf: [u8; 2] = [0; 2];
    let read_status = i2c_handle.read(0x52, &mut read_buf);
```

### PR DESCRIPTION
**Pull request by khrs**
Fix #31

### CODE REVIEW

File: src/i2c.rs [hunk 1]
```diff
@@ -228,12 +228,16 @@ macro_rules! wait_for_flag {
...
+            $i2c.stat0.write(|w| w.berr().clear_bit());
             Err(Other(Error::Bus))
         } else if stat0.lostarb().bit_is_set() {
+            $i2c.stat0.write(|w| w.lostarb().clear_bit());
             Err(Other(Error::Arbitration))
         } else if stat0.aerr().bit_is_set() {
+            $i2c.stat0.write(|w| w.aerr().clear_bit());
             Err(Other(Error::Acknowledge))
         } else if stat0.ouerr().bit_is_set() {
+            $i2c.stat0.write(|w| w.ouerr().clear_bit());
```
Comment (Disasm) [inline src/i2c.rs:240]: I think it's better to use `modify` here. I also asked about this on the corresponding f1xx-hal PR.

**Review by Disasm — COMMENTED** [review]
Thank you! From what I can tell, this PR matches https://github.com/stm32-rs/stm32f1xx-hal/pull/282, so I'm happy to merge it after we get the answer to the `write`/`modify` question.

File: src/i2c.rs (hunk 1 above)
Comment (khrs) [reply src/i2c.rs]: Done

**Review by Disasm — APPROVED** [review]
Thank you!

**Pull request merged**